src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java

```diff
@@ -0,0 +1,178 @@
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;UnnecessarySemicolonAfterTypeMemberDeclaration&quot;/&gt;
+ * </pre>
+ *
+ * @since 8.24
+ */
+@StatelessCheck
+public final class UnnecessarySemicolonAfterTypeMemberDeclarationCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_SEMI = "unnecessary.semicolon";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.ANNOTATION_DEF,
+            TokenTypes.VARIABLE_DEF,
+            TokenTypes.ANNOTATION_FIELD_DEF,
+            TokenTypes.STATIC_INIT,
+            TokenTypes.INSTANCE_INIT,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.ENUM_CONSTANT_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.CLASS_DEF:
+            case TokenTypes.INTERFACE_DEF:
+            case TokenTypes.ENUM_DEF:
+            case TokenTypes.ANNOTATION_DEF:
+                checkTypeDefinition(ast);
+                break;
+            case TokenTypes.VARIABLE_DEF:
+                checkVariableDefinition(ast);
+                break;
+            case TokenTypes.ENUM_CONSTANT_DEF:
+                checkEnumConstant(ast);
+                break;
+            default:
+                checkTypeMember(ast);
+                break;
+        }
+    }
+
+    /**
+     * Checks if type member has unnecessary semicolon.
+     *
+     * @param ast type member
+     */
+    private void checkTypeMember(DetailAST ast) {
+        if (isSemicolon(ast.getNextSibling())) {
+            log(ast.getNextSibling(), MSG_SEMI);
+        }
+    }
+
+    /**
+     * Checks if type definition has unnecessary semicolon.
+     *
+     * @param ast type definition
+     */
+    private void checkTypeDefinition(DetailAST ast) {
+        if (!ScopeUtil.isOuterMostType(ast) && isSemicolon(ast.getNextSibling())) {
+            log(ast.getNextSibling(), MSG_SEMI);
+        }
+        final DetailAST firstMember =
+            ast.findFirstToken(TokenTypes.OBJBLOCK).getFirstChild().getNextSibling();
+        if (isSemicolon(firstMember) && !ScopeUtil.isInEnumBlock(firstMember)) {
+            log(firstMember, MSG_SEMI);
+        }
+    }
+
+    /**
+     * Checks if variable definition has unnecessary semicolon.
+     *
+     * @param ast variable definition
+     */
+    private void checkVariableDefinition(DetailAST ast) {
+        if (isFieldDeclaration(ast) && isSemicolon(ast.getNextSibling())) {
```

Comment:
Method is named `checkVariableDefinition` but first if statement is `isFieldDeclaration`.
Please update method name and javadoc to be field definition.

Reply:
It is correct naming and check. 
`VARIABLE_DEF` token can be local variable declaration or field declaration.
Here we check token and should not check local variables.

Reply:
Yes token is variable definition but the method should be named after what it is doing, not the name of the token it is receiving. If anything `ast` could be renamed to something like `variableDef`.

Reply:
renamed, also isFieldDeclaration method was inlined, please take a look